src/forms/BookingDatesForm/BookingDatesForm.js

```diff
@@ -47,6 +48,25 @@ export class BookingDatesFormComponent extends Component {
     }
   }
 
+  // When the values of the form are updated we need to fetch
+  // lineItems from FTW backend for the EstimatedTransactionMaybe
+  // In case you add more fields to the form, make sure you add
+  // the values here to the bookingData object.
+  handleOnChange(formValues) {
+    const { startDate, endDate } =
+      formValues.values && formValues.values.bookingDates ? formValues.values.bookingDates : {};
+    const listingId = this.props.listingId;
+    const isOwnListing = this.props.isOwnListing;
+
+    if (startDate && endDate && !this.props.fetchLineItemsInProgress) {
```

Comment:
Is it possible to change the values in the form while the line items are being fetched?

If a request takes a long time, and the user can change the dates, the line items won't be fetched for the new dates. In that case the fetch for the latter dates is ignored, and the UI can be in a weird state.

If there isn't a simple solution to cancel/ignore the previous request when a new is being made, then the UI can always prevent selecting new values when the line items are being fetched. Or is it already doing just that?

Reply:
Good catch! I wasn't sure about an easy way to cancel/ignore the previous request so I decided to go with disabling the form while fetch is in progress.